include/deal.II/lac/vector_operations_internal.h

```diff
@@ -1146,7 +1146,15 @@ namespace internal
 
       if (n_chunks > threshold_array_allocate)
         {
-          large_array.resize(n_chunks);
+          if (n_chunks % 2 == 1)
+            {
+              // make sure we allocate an even number of elements,
+              // access to the new last element is needed in do_sum()
+              large_array.resize(n_chunks+1);
+              large_array[n_chunks] = ResultType();
```

Comment:
Why do you need to initialize the last element to zero here? Isn't that already done in line 1176? In case you don't need it, you might be able to save a few lines of code by saying `large_array.resize(2*((n_chunks+1)/2))` that works for both cases.